sip-servlets-client/src/main/java/org/mobicents/javax/servlet/sip/dns/DNSResolver.java

```diff
@@ -60,7 +61,7 @@
 	 * @param uri the uri used to find the corresponding SipURI
 	 * @return the SipURI found through ENUM methods or the uri itself if the uri is already a SipURI without a user=phone param
 	 */
-	SipURI getSipURI(URI uri);
+	SipURI getSipURI(URI uri, SipSession sipSession);
```

Comment:
Hi,
at this class a set of appSessions is defined
https://github.com/RestComm/sip-servlets/blob/master/sip-servlets-impl/src/main/java/org/mobicents/servlet/sip/core/session/SipApplicationSessionCreationThreadLocal.java

But the actual threadlocal reference is hold at
https://github.com/RestComm/sip-servlets/blob/master/containers/sip-servlets-as7/src/main/java/org/mobicents/servlet/sip/startup/SipStandardContext.java#L179

The idea would be to move this threadlocal reference somewhere in
sipServletImpl module, and provide an static access method. So taking the
first appSession in the threadlocal ref, and returning the reference to the
ServletContext. When using threadlocals, its very important to release
references, since they are kept associated in the thread. Since the
threadlocal is required to be static ( so it can be accessed anywhere in
the code with just the proper import) we need to be careful, and consider
service undeploy,redeploy scenarios, and try to gracefully free refs even
in container stop...

Have you ever worked with threadlocals? if not, i may help you with the
threadlocal, and then you can proceed with the rest of the code...

Regards

On Thu, Jun 23, 2016 at 1:24 AM, Luu Huu Hoan <notifications@github.com>
wrote:

> In
> sip-servlets-client/src/main/java/org/mobicents/javax/servlet/sip/dns/DNSResolver.java
> <https://github.com/RestComm/sip-servlets/pull/163#discussion_r68153209>:
>
> > @@ -60,7 +61,7 @@
> >  	 * @param uri the uri used to find the corresponding SipURI
> >  	 * @return the SipURI found through ENUM methods or the uri itself if the uri is already a SipURI without a user=phone param
> >  	 */
> > -	SipURI getSipURI(URI uri);
> > +	SipURI getSipURI(URI uri, SipSession sipSession);
>
> Hi @jaimecasero <https://github.com/jaimecasero> ,
>
> I have spent time to investigate how to use thread local in sipservlet,
> but still don't have any idea. Could you please help me to give me some
> suggestion about this issue.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/RestComm/sip-servlets/pull/163/files/80fac53a2680274d75ce7f27da19ae5f4f488f82#r68153209>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AHe9SX_tXQw_GnYzOMvBS2Mma4t0Nl4Wks5qOcQegaJpZM4Iz60n>
> .
>